### DISCUSSION TITLE
How to create/access connection pool of asyncpg from within Databases?

### DISCUSSION THREAD

**Discussion by houmie** [encode/databases#352] [General]
If I want to create a connection pool for ```asyncpg``` like this

await asyncpg.create_pool(database="dbname", 
    user="username", 
    password="dbpw",
    max_inactive_connection_lifetime=3)

How do I persist and utilise it from within the databases wrapper?  
Thanks

**Comment by houmie** [comment]
@tomchristie 

There is a bug when using ```asyncpg``` with ```Databases``` wrapper.

When too many connections hits the database it throws this error:

```
asyncpg.exceptions.ConnectionDoesNotExistError: connection was closed in the middle of operation
Task was destroyed but it is pending!
```
 
I did quite a bit of research and this can be resolved by defining a pool connection for ```asyncpg``` as explained here: https://stackoverflow.com/questions/56922665/connection-was-closed-in-the-middle-of-operation-when-accesing-database-using-py

and again here in ```asyncpg``` documentation: https://magicstack.github.io/asyncpg/current/usage.html#connection-pools

I have looked at the Databases documentation, but haven't found a way to directly access asyncpg to create such pool.
Can you help proving a hint if Databases supports that? Or should I replace it with plain ```asyncpg``` to have that kind of low level access to connection pool?  

Thanks

**Reply by lovelydinosaur** [reply]
I'm afraid I just don't have the capacity to help with discussions on a one-to-one basis, I've enough on with just `httpx` alone at the moment. Other contributors are very welcome to step up to questions like this.

**Reply by houmie** [reply]
Sure, no worries. Maybe I should have formulated my initial question differently, since an error is thrown, which could be classified as a bug and should be posted in issues.  I don't think anyone pays attention to discussion boards on Github.
